packages/lab/src/checkbox/Checkbox.tsx

```diff
@@ -1,79 +1,169 @@
+  /**
+   * Callback when checkbox loses focus.
+   */
+  onBlur?: FocusEventHandler<HTMLInputElement>;
+  /**
+   * Callback when checked state is changed.
+   */
+  onChange?: (event: ChangeEvent<HTMLInputElement>, checked: boolean) => void;
+  /**
+   * Callback when checkbox gains focus.
+   */
+  onFocus?: FocusEventHandler<HTMLInputElement>;
+  /**
+   * The value of the checkbox.
+   */
+  value?: string;
+}
 
-export const Checkbox = forwardRef<HTMLDivElement, CheckboxProps>(
-  function Checkbox(props, ref) {
-    const {
+export const Checkbox = forwardRef<HTMLLabelElement, CheckboxProps>(
+  function Checkbox(
+    {
       checked: checkedProp,
       className,
       defaultChecked: defaultCheckedProp,
       disabled,
+      error,
       indeterminate,
       inputProps,
       label,
-      LabelProps,
+      name,
+      onBlur,
       onChange,
+      onFocus,
       value,
       ...rest
-    } = props;
-
+    },
+    ref
+  ) {
     const groupContext = useContext(CheckboxGroupContext);
 
+    const handleInternalChange = (event: ChangeEvent<HTMLInputElement>) => {
+      // Workaround for https://github.com/facebook/react/issues/9023
+      if (event.nativeEvent.defaultPrevented) {
+        return;
+      }
+
+      const value = event.target.checked;
+      setChecked(value);
+      onChange?.(event, value);
+    };
+
     const handleChange = createChainedFunction(
-      onChange,
+      handleInternalChange,
       groupContext?.onChange
     );
 
-    let checked = checkedProp;
+    let isCheckchecked = checkedProp;
     let defaultChecked = defaultCheckedProp;
 
     if (groupContext) {
-      if (typeof checked === "undefined" && typeof value === "string") {
-        checked = groupContext?.checkedValues?.includes(value);
+      if (typeof isCheckchecked === "undefined" && typeof value === "string") {
+        isCheckchecked = groupContext?.checkedValues?.includes(value);
       }
-
       defaultChecked = undefined;
     }
 
+    const [checked, setChecked] = useControlled({
+      controlled: isCheckchecked,
+      default: Boolean(defaultCheckedProp),
+      name: "Checkbox",
+      state: "checked",
+    });
+
     return (
-      <div
-        {...rest}
-        className={clsx(classBase, className, {
-          [`${classBase}-disabled`]: disabled,
+      <label
+        className={clsx(withBaseName(), className, {
+          [withBaseName("disabled")]: disabled,
+          [withBaseName("error")]: error,
         })}
-        data-testid="checkbox"
         ref={ref}
+        {...rest}
       >
-        <ControlLabel
-          {...LabelProps}
-          className={`${classBase}-label`}
+        <input
+          aria-checked={indeterminate ? "mixed" : checked}
+          name={name}
+          value={value}
+          {...inputProps}
+          checked={checked}
+          className={withBaseName("input")}
+          data-indeterminate={indeterminate}
+          defaultChecked={defaultChecked}
```

Comment:
I believe we can only pass `checked` attribute to checkbox

Reply:
It's a react specific attribute, when it's rendered, the correct "checked" props is applied